src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let clonesState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clonesState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const n of action.keysToRemove) {
+          delete clonesState[n];
+        }
+        break;
+
+      case 'clear':
+        clonesState = {};
+        break;
```

Comment:
I mentioned before, that default case should handling error, but you just removed it. add default, and add error handling into it
![image](https://user-images.githubusercontent.com/59209337/205855160-32f16c6b-b7ca-49d2-8b7e-dd7fd3a315d0.png)